src/plugins/core/data_validation.ts

```diff
@@ -122,6 +124,24 @@ export class DataValidationPlugin
         this.addDataValidationRule(cmd.sheetId, { ...cmd.rule, ranges });
         break;
       }
+      case "DELETE_CONTENT": {
+        const zones = cmd.target;
+        const sheetId = cmd.sheetId;
+        for (const zone of zones) {
+          for (let row = zone.top; row <= zone.bottom; row++) {
+            for (let col = zone.left; col <= zone.right; col++) {
+              const dataValidation = this.getValidationRuleForCell({ sheetId, col, row });
+              if (!dataValidation || dataValidation.criterion.type !== "isBoolean") {
```

Comment:
functionally is kinda strange that pressing "Delete" removes the checkboxes but not any other DV rule...

Reply:
Gsheet has the same behavior except for the combobox, which should probably be insertable in the same way in another task, so I though we should only keep the checkbox for the moment, and see for the combobox when it will be insertable :)

Reply:
imo, the data validation update upon "delete content" is kind of unrelated to the insertion. If anything it is something that was overlooked in the first implementation and should be added in parallel, but not type by type